app/controllers/partners/family_requests_controller.rb

```diff
@@ -11,19 +11,22 @@ def new
         params[:filterrific]
       ) || return
       @children = @filterrific.find
+      session[:children_ids] = @children.map(&:id)
     end
 
-    def create
-      children_ids = []
+    def confirmation
+      @children = current_partner.children.active.where(id: session[:children_ids]).where.not(item_needed_diaperid: [nil, 0])
 
-      params.each do |key, _|
-        is_child, id = key.split('-')
-        if is_child == 'child'
-          children_ids << id
-        end
+      @items = @children.each_with_object({}) do |child, items|
+        item = Item.find(child.item_needed_diaperid)
+        items[item.name] = {first_name: child.first_name, last_name: child.last_name} unless item.nil?
       end
 
-      children = current_partner.children.active.where(id: children_ids).where.not(item_needed_diaperid: [nil, 0])
+      render :confirmation
+    end
+
+    def create
+      children = current_partner.children.active.where(id: session[:children_ids]).where.not(item_needed_diaperid: [nil, 0])
```

Comment:
Probably should clear the session key after a successful create.

Reply:
added to the end of create: 

```
session.delete(:children_ids)
```